modxcms/revolution
# [Port to 3.x] Fix Various Issues with Form Panel Validation and Field Display - Revised Submission

_NOTE: This PR lays the groundwork for improvements to the element creation UIs, which will be submitted after implementation of this PR. This (and each related PR to follow) is a revised submission of #15887._

### What does it do?
Ports enhancements to the Resource form panel from PR #15146 to 3.x (including subsequent fixes by others). 
Changes include:
1. Added/altered methods in the main FormPanel class to fix issues with/optimize validation — specifically changing more reliably to the (first) errored tab when a field it contains in invalid — which apply to the Resource panel and all Element panels (to come in related PRs momentarily).
2. Addressed a concern raised by @Ruslan-Aleev in #15854 : (7) made style updates so only the settings-type checkboxes in the resource editing form display in the new switch style by default (TVs show the normal checkbox; a new input option setting could be created to control this by assigning the class I added in the forms SCSS ["display-switch"] in a future PR).
3. A couple minor lexicon updates to fix capitalization and make the Summary label read simply “Summary” instead of “Summary (introtext).”

### Why is it needed?
Makes needed UI and consistency improvements to the resource panel.

### How to test

1. From your terminal app, run `grunt build` from within the `_build/templates/default` folder and clear your browser cache.
2. Create at least two TVs (at least one checkbox TV and one TV of any type that is _required_) and verify they operate and appear as expected.
3. Move at least one _required_ TV via Form Customization and assign it to a template. From a resource using this template, verify that (when this TV is not filled and you try to save the resource while viewing a tab other than the one containing this TV) errored tabs are opened and navigated to on save.

### Related issue(s)/PR(s)
Partly addresses #15854 
Partial port of #15146
